### ISSUE TITLE
Generate excel reports

### ISSUE THREAD

**Issue by AntonOellerer** [docu-tools-developers/jocument#14]
The second type of templates to support by jocument are excel documents.
Generally, the placeholders are quite similar to the ones of word documents, with two exceptions:
* Media cannot be integrated 
* Loops are unrolled vertically, but loop content can span multiple columns (see attached image)
![Screenshot 2020-03-17 at 12 23 28](https://user-images.githubusercontent.com/13524304/76854023-a698f000-684e-11ea-80fe-53f4b6ac96dd.png)

On the other hand, excel documents are built up very differently.
This requires thorough studying of the Apache excel API, after which the `ExcelGenerator` should be implemented.
Due to the design of the library, the generator should work with the available resolvers without problems.

Acceptance criteria:
* All previous tests succeed
* All placeholders are resolved successfully
* Integration tests of the excel generator written in the design of the word tests succeed
* Unit tests of the core functions of the excel generator written in the design of the word unit tests succeed.

**Comment by AntonOellerer** [comment]
After reading the [Apache Excel Component](https://poi.apache.org/components/spreadsheet/index.html) documentation, it seems as if there are 2 ways to implement the excel generator:
* ## Usermodel API
Copy the template, use the usermodel API to go through the document and process the placeholders.

**Advantages:**
* Probably easiest way to do
* No problems with references to other cells

**Disadvantages:**
* High memory consumption since the full document is loaded into the memory
* ## Streaming APIs
Use the eventmodel API to read the template and in parallel create the report with the SXSSF API.

**Advantages:**
* Low memory consumption, since only the two APIs windows are loaded into memory

**Disadvantages:**
* Probably more complicated than implementing it with the usermodel API
* Difficult/Impossible to process formulas with references pointing out of the loaded window

**Comment by AntonOellerer** [comment]
**Decision how to move forward:**
Since memory is a big issue when generating larger excel reports, it was decided to for now design the excel generator around the streaming APIs.
2 key insights:
* Design a wrapper around the Apache POI which takes a resolver and interacts with the two APIs in parallel
* Formula evaluation is for now deferred to the person first opening the report, if this introduces unacceptable delay, consider evaluating the formulas on a different server after the report generation

**Comment by AntonOellerer** [comment]
![ExcelGenerator-2](https://user-images.githubusercontent.com/13524304/78234967-39d15700-74d8-11ea-8bfe-a9182f7df287.png)
UML of the first concept

**Comment by AntonOellerer** [comment]
After a first prototype, it was decided to change the design of the Excel resolver.
Moving forward, responsibility is handed back to the `ExcelGenerator`, which navigates each sheet on a row by row, cell by cell basis.

When encountering a loop, it scans forward to find the loop end, and then unrolls it, similarly to how it was done in `DocumentGenerator`.

The main advantage of this is that it is, as mentioned above, similar to the already existing approach, easier to navigate, and can make use of recursion for nested loops.

Due to this, `SAXHandler` and `LoopBuffer` have been removed. 

See 759ce4e08bc5359ed20f89fda100fd7cb46438ec and 8240bd70ab34c7954a3f750b273afc6a6efae51c
